index_batcher.go

```diff
@@ -29,15 +51,10 @@ type IndexBatcher struct {
 	// and adding operations to the batch.
 	lock   sync.Mutex
 	batch  *Batch
-	result *rslt
+	result *error
```

Comment:
I think this can be the much more common and even simpler `error` and not pointer to error.  Error is an interface, so pointers to them are not very common.